File: backend/Controllers/BookingController.cs
using Microsoft.AspNetCore.Mvc;
using backend.Models.Entities;
using backend.Services;

namespace backend.Controllers
{
    [ApiController]
    [Route("api/v1/bookings")]
    public class BookingController : ControllerBase
    {
        private readonly IBookingService _bookingService;

        public BookingController(IBookingService bookingService)
        {
            _bookingService = bookingService;
        }

        [HttpGet]
        public async Task<ActionResult<List<Booking>>> GetAllBookings()
        {
            var bookings = await _bookingService.GetAllBookings();
            return Ok(bookings);
        }

        [HttpGet("client/{clientId}")]
        public async Task<ActionResult<List<Booking>>> GetBookingsByClient(long clientId)
        {
            var bookings = await _bookingService.GetBookingsByClient(clientId);
            return Ok(bookings);
        }

        [HttpGet("vendor/{vendorId}")]
        public async Task<ActionResult<List<Booking>>> GetBookingsByVendor(long vendorId)
        {
            var bookings = await _bookingService.GetBookingsByVendor(vendorId);
            return Ok(bookings);
        }

        [HttpPost]
        public async Task<ActionResult<Booking>> CreateBooking([FromBody] Booking booking)
        {
            var createdBooking = await _bookingService.CreateBooking(booking);
            return Ok(createdBooking);
        }

        [HttpPatch("{id}/status")]
        public async Task<ActionResult<Booking>> UpdateBookingStatus(long id, [FromBody] Dictionary<string, string> statusMap)
        {
            if (statusMap.TryGetValue("status", out var status))
            {
                var updatedBooking = await _bookingService.UpdateBookingStatus(id, status);
                if (updatedBooking == null) return NotFound();
                return Ok(updatedBooking);
            }
            return BadRequest("Status is required");
        }
    }
}

File: springboot-backend/src/main/java/com/wedspot/backend/controller/BookingController.java
package com.wedspot.backend.controller;

import com.wedspot.backend.Model.Entity.Booking;
import com.wedspot.backend.services.IBookingService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/bookings")
@RequiredArgsConstructor
public class BookingController {

    private final IBookingService bookingService;

    @GetMapping
    public ResponseEntity<List<Booking>> getAllBookings() {
        return ResponseEntity.ok(bookingService.getAllBookings());
    }

    @GetMapping("/client/{clientId}")
    public ResponseEntity<List<Booking>> getBookingsByClient(@PathVariable Long clientId) {
        return ResponseEntity.ok(bookingService.getBookingsByClient(clientId));
    }

    @GetMapping("/vendor/{vendorId}")
    public ResponseEntity<List<Booking>> getBookingsByVendor(@PathVariable Long vendorId) {
        return ResponseEntity.ok(bookingService.getBookingsByVendor(vendorId));
    }

    @PostMapping
    public ResponseEntity<Booking> createBooking(@RequestBody Booking booking) {
        return ResponseEntity.ok(bookingService.createBooking(booking));
    }

    @PatchMapping("/{id}/status")
    public ResponseEntity<Booking> updateBookingStatus(@PathVariable Long id, @RequestBody Map<String, String> statusMap) {
        return bookingService.updateBookingStatus(id, statusMap.get("status"))
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }
}
